src/main/webapp/app/exam/manage/exams/exam-checklist-component/exam-checklist.component.ts

```diff
@@ -62,9 +62,13 @@ export class ExamChecklistComponent implements OnInit {
         if (this.pointsExercisesEqual) {
             this.exam.exerciseGroups!.forEach((exerciseGroup) => {
                 if (exerciseGroup.isMandatory) {
-                    sumPointsExerciseGroupsMandatory += exerciseGroup!.exercises![0]!.maxPoints!;
+                    if (exerciseGroup!.exercises && exerciseGroup.exercises.length !== 0) {
+                        sumPointsExerciseGroupsMandatory += exerciseGroup!.exercises![0]!.maxPoints!;
+                    }
                 } else {
-                    sumPointsExerciseGroupsOptional += exerciseGroup!.exercises![0]!.maxPoints!;
+                    if (exerciseGroup!.exercises && exerciseGroup.exercises.length !== 0) {
+                        sumPointsExerciseGroupsOptional += exerciseGroup!.exercises![0]!.maxPoints!;
+                    }
```

Comment:
you could put this: `                     if (exerciseGroup!.exercises && exerciseGroup.exercises.length !== 0) {` outside of the `                 if (exerciseGroup.isMandatory) {`, then you wouldn't have to use the same line twice.

Reply:
good catch, that looks better.